js/private/node-patches/src/fs.ts

```diff
@@ -621,19 +642,16 @@ export const patcher = (fs: any = _fs, roots: string[]) => {
         function _tryParent(): boolean {
             nested.push(path.basename(maybe))
             const dirname = path.dirname(maybe)
-            if (
-                !dirname ||
-                dirname == maybe ||
-                dirname == '.' ||
-                dirname == '/'
-            ) {
+            // TODO: also look for paths that are not within any 'roots'
+            if (dirname == maybe || HOP_ROOTS[dirname] || HOP_ROOTS[maybe]) {
```

Comment:
This essentially does the same as before, but also stops when hitting `/var/.../bazel-out` or I think there's always (often?) also `/var/.../bazel-out/.../target.runfiles` which will be hit first.

Note that if something the `dirname` or `maybe` skip those paths this does not catch paths *above* those roots which is essentially what the TODO above is.